src/server/main.js

```diff
@@ -83,6 +83,22 @@ io.on('connect', (socket) => {
     return socket.emit('uploadMemeData', saveResult);
   });
 
+  socket.on('addComment', async (data) => {
+    if (socketUser === false) {
+      return socket.emit('addComment', 'cannot verify user');
+    }
+
+    const commentResult = await meme.addComment(data, socketUser);
+
+    return socket.emit('addComment', commentResult);
```

Comment:
Shouldn't this be io.emit, so it can be broadcasted to all users viewing the meme?

Reply:
Yes it is ! Good call.

Reply:
Wait no, this event only adds a comment, it returns if commenting was successful or not.

Reply:
It adds a comment and if it was successful, it should broadcast into a room for those who are viewing the meme at the moment, as well as tell the commenter it was successful. There will need to be other events for entering and leaving the rooms. One room per meme.

Reply:
Joining a room could be part of getMemeComments

Reply:
@nxtman123 I'm not quite sure by what you mean by having events for entering/leaving rooms.
Could you explain that some more?

Reply:
Do you mean like having an event for viewing the memes?

ie. the data being passed to the event would be the meme_id, then it would return the contents of the meme associated with that meme_id?

Reply:
yup, that would work well. that handler would call `socket.join('meme-42') to join the room while it's returning the meme details. Another event would be needed for when the client navigates away from the meme page to call `socket.leave('meme-42')`.

Also, getting the existing memes (not the ones that are added while on the page) should have pagination, so that only the first (idk, 30) comments are loaded at first and then more can be loaded later (using the infinite scroll on the comments).

Reply:
Alright, that's more clear.

So in this pull request: 
* I will modify the function so that it pushes the new comments to all the users viewing the certain meme and
* Modify the retrieve comments function so that it only retrieves the latest 30 comments